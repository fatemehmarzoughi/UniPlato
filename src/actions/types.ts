/* -------------------------------------------------------------------------- */
/*                             Posts Action Types                             */
/* -------------------------------------------------------------------------- */
export const GET_ALL_POSTS: string = 'GET_ALL_POSTS';
export const GET_ALL_POSTS_SUCCESS: string = 'GET_ALL_POSTS_SUCCESS';
export const GET_ALL_POSTS_FAILED: string = 'GET_ALL_POST_FAILED';