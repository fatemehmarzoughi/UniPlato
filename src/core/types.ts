import { PostStates } from "../reducer/postsReducer"

export type Post = {
    id: number,
    title: string,
    description: string,
    image: string,
}

export type AppStates = {
    postsStates: PostStates,
}