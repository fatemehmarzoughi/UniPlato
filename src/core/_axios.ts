import axios from 'axios';
import { BASE_URL } from '@src/constants/general';

const baseUrl = axios.create({
    baseURL: BASE_URL
})

export default baseUrl;