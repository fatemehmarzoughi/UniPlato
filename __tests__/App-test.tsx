import * as React from "react";
import renderer from "react-test-renderer";
import Card from "../src/components/card";
import { Post } from "../src/core/types";
import Button from '../src/components/button';

it(`renders Card component correctly with the post's details informations`, () => {
  const item: Post = {
    id: 0,
    title: 'title',
    description: 'description',
    image: 'imageURL'
  }
  const CardSnapShot = renderer.create(<Card item={item} />);
  expect(CardSnapShot).toMatchSnapshot();
  expect(CardSnapShot.root.props.item.id).toBe(0);
  expect(CardSnapShot.root.props.item.title).toBe('title');
  expect(CardSnapShot.root.props.item.description).toBe('description');
  expect(CardSnapShot.root.props.item.image).toBe('imageURL');
});

it(`renders Button component correctly with the props's details informations`, () => {

  const buttonSnapShot = renderer.create(<Button buttonText="buttonText" onPress={() => {}} />);
  expect(buttonSnapShot).toMatchSnapshot();
  expect(buttonSnapShot.root.props.buttonText).toBe('buttonText')

});