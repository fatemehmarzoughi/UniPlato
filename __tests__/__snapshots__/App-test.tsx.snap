// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Button component correctly with the props's details informations 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#B3EBF7",
      "borderRadius": 10,
      "margin": 10,
      "opacity": 1,
      "padding": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#00BEDF",
      }
    }
  >
    buttonText
  </Text>
</View>
`;

exports[`renders Card component correctly with the post's details informations 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 20,
      "margin": 10,
      "marginBottom": 25,
      "padding": 20,
      "paddingBottom": 25,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "imageURL",
      }
    }
    style={
      Object {
        "borderRadius": 25,
        "height": 690,
        "marginBottom": 30,
        "width": 690,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    title
  </Text>
  <Text
    numberOfLines={3}
    style={
      Object {
        "color": "gray",
        "fontSize": 15,
      }
    }
  >
    description
  </Text>
</View>
`;

exports[`renders card component correctly with the post's details informations 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 20,
      "margin": 10,
      "marginBottom": 25,
      "padding": 20,
      "paddingBottom": 25,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "imageURL",
      }
    }
    style={
      Object {
        "borderRadius": 25,
        "height": 690,
        "marginBottom": 30,
        "width": 690,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    title
  </Text>
  <Text
    numberOfLines={3}
    style={
      Object {
        "color": "gray",
        "fontSize": 15,
      }
    }
  >
    description
  </Text>
</View>
`;

exports[`renders card component correctly with the post's details informations 2`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "backgroundColor": "#B3EBF7",
      "borderRadius": 10,
      "margin": 10,
      "opacity": 1,
      "padding": 15,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#00BEDF",
      }
    }
  >
    buttonText
  </Text>
</View>
`;

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "backgroundColor": "white",
      "borderRadius": 20,
      "margin": 10,
      "marginBottom": 25,
      "padding": 20,
      "paddingBottom": 25,
    }
  }
>
  <Image
    source={
      Object {
        "uri": "imageURL",
      }
    }
    style={
      Object {
        "borderRadius": 25,
        "height": 690,
        "marginBottom": 30,
        "width": 690,
      }
    }
  />
  <Text
    style={
      Object {
        "fontSize": 20,
        "fontWeight": "bold",
        "marginBottom": 10,
      }
    }
  >
    title
  </Text>
  <Text
    numberOfLines={3}
    style={
      Object {
        "color": "gray",
        "fontSize": 15,
      }
    }
  >
    description
  </Text>
</View>
`;
